worker/upgradesteps/manifold.go

```diff
@@ -56,32 +58,39 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 			// the Engine to abort us when conn is closed...
 			var apiConn api.Connection
 			if err := context.Get(config.APICallerName, &apiConn); err != nil {
-				return nil, err
+				return nil, errors.Trace(err)
 			}
 
 			// Get upgradesteps completed lock.
 			var upgradeStepsLock gate.Lock
 			if err := context.Get(config.UpgradeStepsGateName, &upgradeStepsLock); err != nil {
-				return nil, err
+				return nil, errors.Trace(err)
 			}
 
 			// Get the agent's jobs.
 			// TODO(fwereade): use appropriate facade!
-			agentFacade, err := apiagent.NewState(apiConn)
-			if err != nil {
-				return nil, err
-			}
-			entity, err := agentFacade.Entity(agent.CurrentConfig().Tag())
-			if err != nil {
-				return nil, err
+			var jobs []multiwatcher.MachineJob
+			if names.IsValidApplication(agent.CurrentConfig().Tag().String()) {
+				// application tag for CAAS operator.
+				jobs = []multiwatcher.MachineJob{}
```

Comment:
We don't need an empty slice do we?
Can't we just leave it as nil? If not let's fix it so we can and avoid the multiwatcher import.

Reply:
sure, fixed,